pinot-core/src/main/java/org/apache/pinot/core/operator/streaming/StreamingSelectionOnlyCombineOperator.java

```diff
@@ -47,13 +47,17 @@ public String toExplainString() {
   }
 
   @Override
-  protected boolean isOperatorSatisfied(SelectionResultsBlock resultsBlock) {
-    long numRowsCollected = _numRowsCollected.addAndGet(resultsBlock.getNumRows());
-    return numRowsCollected >= _limit;
+  protected boolean isChildOperatorSingleBlock() {
+    return false;
```

Comment:
Shouldn't this return to false at `BaseStreamingCombineOperator` level itself assuming single block here non-streaming combine operator that merge into a single block ?

Reply:
This is for the child operator (e.g. AggregationOperator) which is not the CombineOperator. Most of the operators (executor on a single segment) returns a single block, with the only exception of `StreamingSelectionOnlyOperator`.